pkg/storage/unified/apistore/restore.go

```diff
@@ -0,0 +1,102 @@
+package apistore
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/registry/generic"
+	"k8s.io/apiserver/pkg/registry/rest"
+
+	"github.com/grafana/grafana/pkg/storage/unified/resource"
+)
+
+type RestoreConnector interface {
+	rest.Storage
+	rest.Connecter
+	rest.StorageMetadata
+}
+
+func NewRestoreConnector(unified resource.ResourceClient, gr schema.GroupResource, opts generic.RESTOptions) RestoreConnector {
+	return &restoreREST{
+		unified: unified,
+		gr:      gr,
+		opts:    opts,
+	}
+}
+
+type restoreREST struct {
+	unified resource.ResourceClient
+	gr      schema.GroupResource
+	opts    generic.RESTOptions
+}
+
+func (r *restoreREST) New() runtime.Object {
+	return &metav1.PartialObjectMetadataList{}
+}
+
+func (r *restoreREST) Destroy() {
+}
+
+func (r *restoreREST) ConnectMethods() []string {
+	return []string{"POST"}
+}
+
+func (r *restoreREST) ProducesMIMETypes(verb string) []string {
+	return nil
+}
+
+func (r *restoreREST) ProducesObject(verb string) interface{} {
+	return &metav1.PartialObjectMetadataList{}
+}
+
+func (r *restoreREST) NewConnectOptions() (runtime.Object, bool, string) {
+	return nil, false, ""
+}
+
+func (r *restoreREST) Connect(ctx context.Context, uid string, opts runtime.Object, responder rest.Responder) (http.Handler, error) {
+	info, err := NamespaceInfoFrom(ctx, true)
+	if err != nil {
+		return nil, err
+	}
+
+	key := &resource.ResourceKey{
+		Namespace: info.Value,
+		Group:     r.gr.Group,
+		Resource:  r.gr.Resource,
+		Name:      uid,
+	}
+
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		query := req.URL.Query()
+		res := query.Get("resourceVersion")
+		rv, err := strconv.ParseInt(res, 10, 64)
+		if err != nil || rv == 0 {
```

Comment:
I think `rv == 0` might still be useful, e.g. when restoring a deleted dashboard to the latest known version.

The client might not necessarily know which specific version that was.

Or do you envision this to be done on the client side by requesting the `/latest` first, extracting the RV from it and then POSTing that to `/restore`?

Reply:
> Or do you envision this to be done on the client side by requesting the /latest first, extracting the RV from it and then POSTing that to /restore?

This :) but this is more of a POC for now, would like to get feedback on the APIs from the frontend and go from there, if that is okay with you?